lib/sql.js

```diff
@@ -403,42 +1167,151 @@ SqlConnector.prototype.automigrate = function (models, cb) {
 };
 
 /**
- * Drop the table for the given model from the database
- * @param {String} model The model name
- * @param {Function} [cb] The callback function
+ * Serialize an object into JSON string or other primitive types so that it
+ * can be saved into a RDB column
+ * @param {Object} obj The object value
+ * @returns {*}
  */
-SqlConnector.prototype.dropTable = function (model, cb) {
-  this.command('DROP TABLE IF EXISTS ' + this.tableEscaped(model), cb);
+SqlConnector.prototype.serializeObject = function(obj) {
+  var val;
+  if (obj && typeof obj.toJSON === 'function') {
+    obj = obj.toJSON();
+  }
+  if (typeof obj !== 'string') {
+    val = JSON.stringify(obj);
+  } else {
+    val = obj;
+  }
+  return val;
+};
+
+/*!
+ * @param obj
+ */
+SqlConnector.prototype.escapeObject = function(obj) {
+  var val = this.serializeObject(obj);
+  return this.escapeValue(val);
 };
 
 /**
- * Create the table for the given model
- * @param {String} model The model name
- * @param {Function} [cb] The callback function
+ * The following _abstract_ methods have to be implemented by connectors that
+ * extend from SqlConnector to reuse the base implementations of CRUD methods
+ * from SqlConnector
  */
 
-SqlConnector.prototype.createTable = function (model, cb) {
-  this.command('CREATE TABLE ' + this.tableEscaped(model) +
-    ' (\n  ' + this.propertiesSQL(model) + '\n)', cb);
+/**
+ * Converts a model property value into the form required by the
+ * database column. The result should be one of following forms:
+ *
+ * - {sql: "point(?,?)", params:[10,20]}
+ * - {sql: "'John'", params: []}
+ * - "John"
+ *
+ * @param {Object} propertyDef Model property definition
+ * @param {*} value Model property value
+ * @returns {ParameterizedSQL|*} Database column value.
+ *
+ */
+SqlConnector.prototype.toColumnValue = function(propertyDef, value) {
+  /*jshint unused:false */
```

Comment:
AFAICT, this jshint directive is not needed, because it's already set in the master jshintrc file. This comment applies to other abstract methods too.